tests/CMakeLists.txt

```diff
@@ -128,7 +128,8 @@ set(PYBIND11_TEST_FILES
     test_custom_type_casters
     test_custom_type_setup
     test_docstring_options
-    test_eigen
+    test_eigen_tensor
```

Comment:
Could you please swap these two, to keep this list in alphabetical order?

Also for the blocks at lines 237 and 296, could those be shuffled, too?